contracts/GroupVoteLocker.sol

```diff
@@ -0,0 +1,331 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "./BaseVoteLocker.sol";
+import "OpenZeppelin/openzeppelin-contracts@02fcc75bb7f35376c22def91b0fb9bc7a50b9458/contracts/utils/math/SafeCast.sol";
+import "OpenZeppelin/openzeppelin-contracts@02fcc75bb7f35376c22def91b0fb9bc7a50b9458/contracts/utils/Strings.sol";
+import "./console.sol";
+
+contract GroupVoteLocker is BaseVoteLocker {
+    ///@notice Checkpoint structure representing linear voting power decay
+    struct Checkpoint {
+        int128 bias;
+        int128 slope;
+        uint256 ts;
+        uint256 blk;
+    }
+
+    ///@notice Stores token lockup details
+    struct Lockup {
+        int128 amount;
```

Comment:
Why is Lockup.amount an int128? Can people do negative lockups?

Reply:
This was taken straight from the Curve contract. I didn't attempt to change the structure.